From 100a0351bdeb68e8711d5c6d57591053877e1e36 Mon Sep 17 00:00:00 2001
From: Guanhua Gao <guanhua.gao@nxp.com>
Date: Tue, 15 May 2018 16:00:08 +0800
Subject: [PATCH 507/767] staging: fsl-dpaa2/eth: Sync with latest dpaa2
 drivers

Sync with latest dpaa2 drivers.

Signed-off-by: Guanhua Gao <guanhua.gao@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index fdc343e79abb..38a3efd66f22 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -258,7 +258,8 @@ static int dpaa2_eth_xdp_tx(struct dpaa2_eth_priv *priv,
 
 	fq = &priv->fq[queue_id];
 	for (i = 0; i < DPAA2_ETH_ENQUEUE_RETRIES; i++) {
-		err = dpaa2_io_service_enqueue_qd(NULL, priv->tx_qdid, 0,
+		err = dpaa2_io_service_enqueue_qd(fq->channel->dpio,
+						  priv->tx_qdid, 0,
 						  fq->tx_qdbin, fd);
 		if (err != -EBUSY)
 			break;
@@ -283,12 +284,12 @@ static void release_fd_buf(struct dpaa2_eth_priv *priv,
 	if (likely(ch->rel_buf_cnt < DPAA2_ETH_BUFS_PER_CMD))
 		return;
 
-	while ((err = dpaa2_io_service_release(NULL, priv->bpid,
+	while ((err = dpaa2_io_service_release(ch->dpio, priv->bpid,
 					       ch->rel_buf_array,
 					       ch->rel_buf_cnt)) == -EBUSY)
  		cpu_relax();
  
- 	if (unlikely(err))
+	if (err)
 		free_bufs(priv, ch->rel_buf_array, ch->rel_buf_cnt);
  
 	ch->rel_buf_cnt = 0;
@@ -3916,11 +3917,11 @@ static int dpaa2_eth_remove(struct fsl_mc_device *ls_dev)
 #endif
 	dpaa2_eth_sysfs_remove(&net_dev->dev);
 
+	unregister_netdev(net_dev);
+
 	disable_ch_napi(priv);
 	del_ch_napi(priv);
 
-	unregister_netdev(net_dev);
-
 	if (priv->do_link_poll)
 		kthread_stop(priv->poll_thread);
 	else
-- 
2.17.0

